components/Popper/usePopper.js

```diff
@@ -0,0 +1,61 @@
+import { useState, useEffect } from 'react';
+import { adjustPosition, computedSize } from './utils';
+
+const usePopper = (
+    popperRef,
+    anchorRef,
+    visible,
+    { originalPlacement = 'bottom', offset = 10, scrollContainerClass = null } = {}
+) => {
+    const [placement, setPlacement] = useState(originalPlacement);
+    const [position, setPosition] = useState({ top: -1000, left: -1000 });
+
+    useEffect(() => {
+        const updatePosition = () => {
```

Comment:
Is it necessary to put this function in `useEffect`? I prefer to move it outside if possible.

Reply:
I prefer to keep it inside because then it makes it clearer when it's run and the function is only created when it's needed

Reply:
if it was outside, I would need to wrap it in useCallback, which makes it uglier :D